### PR TITLE
Begin adding checkpointing support for Activity Classification

### PR DESCRIPTION
**Pull request by nickjong**

### CODE REVIEW

File: src/toolkits/activity_classification/activity_classifier.cpp [hunk 1]
```diff
@@ -392,13 +409,7 @@ void activity_classifier::train(
   }
 
   state_update["verbose"] = read_state<bool>("verbose");
-  state_update["num_examples"] = data.size();
```
Comment (nickjong) [inline src/toolkits/activity_classification/activity_classifier.cpp]: Note that before we were giving the number of examples in the `data` argument. If the validation argument was `"auto"` then this included the validation set too. I fixed this along the way, so now we already write the actual training set size.
